File: recipe-backend/Models/Recipe.cs
using System.ComponentModel.DataAnnotations;

namespace Models
{
    public class Recipe
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string[] Description { get; set; }
        [Required]
        public string[] Ingredients { get; set; }
        public string[] Categories { get; set; }
        public double? Calories { get; set; }
        public int? Servings { get; set; }
        public NutritionInfo NutritionInfo { get; set; }
        public int? DurationInMinutes { get; set; }
    }
}

File: recipe-frontend/src/app/models/recipe.ts
import { Difficulty } from "./difficulty";
import { NutritionInfo } from "./nutritionInfo";

export class Recipe {
    id: number | undefined;
    name?: string | undefined;
    description?: string[] | undefined;
    ingredients?: string[] | undefined;
    categories?: string[] | undefined;
    calories?: number | undefined;
    servings?: number | undefined;
    durationInMinutes?: number | undefined;
    nutritionInfoId?: number | null;
    nutritionInfo?: NutritionInfo;
    difficulty: Difficulty = Difficulty.None;
    addedByUser: boolean = false;
    image?: string | undefined;
    imgaeId?: number | null;
    ingredientsFound : string[] = [];
}
